### ISSUE TITLE
[Proposal] attachmentbrowser button to copy path to clipboard

### ISSUE THREAD

**Issue by jkooij** [zim-desktop-wiki/zim-desktop-wiki#1347]
## Summary
Add a menu button to the attachment browser to copy the current attachment folder path to the clipboard.
I've already created a patch and will add a pull request.

## Use case
One of my most common use cases for Zim is to collect and organize related document as attachments to a page. E.g. often I get new document versions send via email, or I download pdfs via the browser, and I want to store these together with the relevant page in my notebook. In all these cases, I need to store the documents from some external GUI program, and I want to save them into the directory created by Zim's attachment browser. This requires copying the attachment folder path to the "Save As" popup window of the external program.

## Before proposed change
My usual workflow to save a document to the attachment folder is as follows:
1. Start external program X's option to export the document(s), e.g. press some "Save As" button
2. X opens a "Save As" popup window to select the destination path. Often this popup is a native OS window, and enables the user to navigate to a desired folder, but this takes a lot of clicking and searching. The popup often also has an address bar.
3. Click on Zim's attachment browser's "Open Folder" menu button (and use option to create it if it doesn't exist). This opens the folder in the native file explorer.
4. In the file explorer, select the address bar containing the full directory path
5. press Ctrl-A to select the full path
6. press Ctrl-C to copy to the clipboard
7. Click on address bar of the SaveAs popup from program X
8. press Ctrl-V + Enter to replace the path with the folder from the attachment browser
9. Press Enter or the Save button in the popup window to store the document in the attachment folder
10. Close the opened file explorer window

This workflow is not very efficient. Notice that the file explorer was only opened as a means to get the folder path in the clipboard, and should be closed in the final step to avoid cluttering my work environment.

## After proposed change
I propose to add a third button "Copy path" to the attachment browser. This would **replace steps 4, 5, 6 and 10, and the need to wait for a file explorer window to open and close.**

1. Start external program X's option to export the document(s), e.g. press some "Save As" button
2. X opens a "Save As" popup window to select the destination path. Often this popup is a native OS window, and enables the user to navigate to a desired folder, but this takes a lot of clicking and searching. The popup often also has an address bar.
3. Click on Zim's attachment browser's "Copy Path" menu button (and use option to create it if it doesn't exist). This copies the folder path to the clipboard.
4. Click on address bar of the SaveAs popup from program X
5. press Ctrl-V + Enter to replace the path with the folder from the attachment browser
6. Press Enter or the Save button in the popup window to store the document in the attachment folder

Overall, a small and simple change can simplify this common workflow and save time and window clutter.

**Comment by pkral78** [comment]
Yeah but it is still point'n'click porn, less hardcore of course. I'm thinking more like about some 'INBOX' folder, where attachments could be saved as quickly as possible (that folder could be added to the 'Favorites' file browser shortcut or just simple copy CLI may be used). This then requires to create dialog that manage recently added files from 'INBOX' folder and moves files to the note attachment folder.

**Comment by jaap-karssenberg** [comment]
On the one hand I really see how this feature solves your problem, on the other hand I'm not sure this feature makes as a generic feature for all users and reluctant to add it to the attachment browser buttonbox.

I think you achieve the same behavior with a "custom tool" since it allows to add (external) scripts to the toolbar and set it up to take the attachment folder as an input parameter. Would that work for you ?

**Comment by introt** [comment]
For a more generic feature, adding "copy path to file" to the right-click menu of attachments would both facilitate attachment reuse in multiple pages and ease this person's workflow: they could then just edit the filename part of the pasted path and press enter to save.
